traces/metrics.go

```diff
@@ -180,9 +182,10 @@ func Metrics(csvFile string, filenameColumn string, sink MetricsSink) error {
 			}
 
 			return nil
-		})
+		}
 
-		if err != nil {
+		if err := readLargeCsvFile(csvFile,
+			tolerateFaults(csvFile, emitMetricsFromRow)); err != nil {
 			return err
 		}
```

Comment:
Same suggestion here:
```suggestion
                tolerantReader = tolerateFaults(csvFile, emitMetricsFromRow)
		if err := readLargeCsvFile(csvFile, tolerantReader); err != nil {
			return err
		}
```